cmd/limactl/shell.go

```diff
@@ -151,7 +151,20 @@ func shellAction(cmd *cobra.Command, args []string) error {
 		"--",
 		script,
 	}...)
-	sshCmd := exec.Command(arg0, sshArgs...)
+
+	var sshCmd *exec.Cmd
+
+	if usrShell := os.Getenv(envShell); isTerm && usrShell != "" {
```

Comment:
Doesn't `$SHELL` work when `!isTerm` ?

Reply:
It does, but what would be the benefit of shelling out if not in a terminal?

Reply:
I thought the aliases are expected to be honored regardless to presence of tty

Reply:
There are some ramifications to consider:

- `.<X>sh_profile` is only read if the shell is a _login_ shell
- `.<X>shrc` is only read if the shell is an _interactive_ shell

Omitting `-l` makes shelling out useless, since neither `.<X>sh_profile` nor `.<X>shrc` are considered in this case.
So we _always_ need to pass `-l` to make the feature useful.

Setting `-i` without a tty prints an error message, so we shouldn't set in in the absence of tty.

Are we OK with this?

Reply:
Sorry to go back and forth, but maybe just support custom ssh command via `$SSH` might be better?
An equivalent of the option 3 in the original issue but with a simpler env var name.

Reply:
(Ref. #1132 for completeness)

Reply:
I like both approaches equally really.
The env var offers control, the shell makes things work without extra config.

@jandubois what's your opinion?

Reply:
Why an env-var and not just a config option in the lima.yaml? I could see either/both making sense.

Reply:
> The env var offers control, the shell makes things work without extra config.
> 
> @jandubois what's your opinion?

Given the issues we ran into making it work with different shells, login shells vs. interactive shells, and how they are treated differently on macOS vs. Linux, I think I prefer adding the `LIMA_SSH` variable.

I think I prefer `LIMA_SSH` over `SSH`, but probably can be convinced otherwise. 😄 

> Why an env-var and not just a config option in the lima.yaml? I could see either/both making sense.

I do not view this as an instance-specific setting, so I think `lima.yaml` is the wrong place for it. Why would you want different `ssh` commands for different instances?

Reply:
Agree about this not being instance specific.

I would prefer to avoid `SSH` as well, this is too generic, not in Lima's namespace.

I like `LIMASHELL_SSH` better than other options because it tells exactly what command it relates too, whereas `LIMA_` could have a very broad scope.

I'm re-opening #1132.